static/scripts/files-storage.js

```diff
@@ -0,0 +1,280 @@
+				fileName,
+			},
+			url: `/api/v3/file/rename/${fileRecordId}`,
+			type: 'PATCH',
+			success: () => reloadPage(undefined, 0),
+		}).fail(showAJAXError);
+	}
+
+	function updateUploadProcessingProgress() {
+		if (!progressBarActive) {
+			$progress.css('width', '0%');
+
+			$form.fadeOut(50, () => {
+				$progressBar.fadeIn(50);
+			});
+
+			progressBarActive = true;
+		}
+	}
+
+	if ($form.dropzone) {
+		$form.dropzone({
+			accept(file, done) {
+				const fdata = new FormData();
+				fdata.append('file', file);
+				$.ajax({
+					cache: false,
+					data: fdata,
+					url: `/api/v3/file/upload/
+					${getSchoolId()}/
+					${getCurrentParentType()}/
+					${getCurrentParentId()}`,
+					type: 'POST',
+					processData: false,
+					contentType: false,
+				})
+					.done(() => {
+						done();
+					}).fail(showAJAXError);
+			},
+			createImageThumbnails: false,
+			method: 'PUT',
+			init() {
+				// this is called on per-file basis
+				this.on('processing', updateUploadProcessingProgress);
+
+				this.on('sending', (file, xhr) => {
+					const { send } = xhr;
+					xhr.send = () => {
+						send.call(xhr, file);
+					};
+				});
+
+				this.on('totaluploadprogress', (_, total, uploaded) => {
+					const realProgress = (uploaded + finishedFilesSize) / ((total + finishedFilesSize) / 100);
+
+					$progress.stop().animate(
+						{ width: `${realProgress}%` },
+						{
+							step(now) {
+								if ($percentage && $percentage.setAttribute) {
+									$percentage.html(`${Math.ceil(now)}%`);
+									$percentage.setAttribute('aria-valuenow', `${Math.ceil(now)}%`);
+								}
+							},
+						},
+					);
+				});
+
+				this.on('queuecomplete', () => {
+					progressBarActive = false;
+					finishedFilesSize = 0;
+
+					$progressBar.fadeOut(50, () => {
+						$form.fadeIn(50);
+						reloadPage('files._file.text.fileAddedSuccess');
+					});
+				});
+
+				this.on('dragover', () => $form.addClass('focus'));
+				this.on('dragleave', () => $form.removeClass('focus'));
+				this.on('dragend', () => $form.removeClass('focus'));
+				this.on('drop', () => $form.removeClass('focus'));
+			},
+		});
+	}
+
+	$('button[data-method="download"]').on('click', (e) => {
+		window.open($(e.currentTarget).attr('data-href'), '_blank');
+		e.stopPropagation();
+	});
+
+	$modals.find('.close, .btn-close').on('click', () => {
+		$modals.modal('hide');
+	});
+
+	$('.btn-file-danger').click((e) => {
+		e.stopPropagation();
+		e.preventDefault();
+		const $dangerModal = $('.danger-modal');
+		$dangerModal.appendTo('body').modal('show');
+	});
+
+	$($renameModal.find('.btn-submit')).click((e) => {
+		e.stopPropagation();
+		e.preventDefault();
+		const fileRecordId = $renameModal.find('#fileRecordId').val();
+		const fileName = $renameModal.find('#newNameInput').val();
+		rename(fileName, fileRecordId);
+		$renameModal.modal('hide');
+	});
+
+	$('#files-storage-component').find('button[data-method="delete"]').on('click', deleteFileClickHandler);
```

Comment:
Sometimes we use .click( and somethimes on('click',  ..i think the second is the newer version without waring or?
Then we should use only this without warning.